action-server/covidflow/utils/screening_locations.py

```diff
@@ -0,0 +1,66 @@
+from typing import Iterable
+
+import backoff
+import structlog
+from aiohttp import ClientError, ClientSession
+
+logger = structlog.get_logger()
+
+CLINIA_ENDPOINT = "https://covid.clinia.com"
+CLINIA_API_ROUTE = "/api/v1/indexes/covid/query"
+
+LOCATION_KEY = "aroundLatLng"
+PAGE_KEY = "page"
+HITS_KEY = "hits"
+
+DEFAULT_SEARCH_PARAMETERS = {
+    "perPage": 20,
+    "rankingInfo": True,
+    "facetFilters": ["services.en:COVID-19 screening"],
+}
+
+HTTP_OK = 200
+
+
+async def _fetch_screening_locations(
+    session: ClientSession, latitude: float, longitude: float, page: int = 0
+):
+    body = {
+        **DEFAULT_SEARCH_PARAMETERS,
+        LOCATION_KEY: f"{latitude},{longitude}",
+        PAGE_KEY: page,
+    }
+
+    url = f"{CLINIA_ENDPOINT}{CLINIA_API_ROUTE}"
+    logger.debug(f"Fetching test sites", url=url, body=body)
+
+    result = await session.post(url, json=body)
+    return await result.json()
+
+
+@backoff.on_exception(backoff.expo, ClientError, max_time=3)
+async def _fetch_screening_locations_with_backoff(
+    session: ClientSession, latitude: float, longitude: float, page: int = 0
+):
+    return await _fetch_screening_locations(
+        session=session, latitude=latitude, longitude=longitude, page=page
+    )
+
+
+async def get_screening_locations(latitude: float, longitude: float) -> Iterable[dict]:
+    async with ClientSession(raise_for_status=True) as session:
+        result = await _fetch_screening_locations_with_backoff(
+            session=session, latitude=latitude, longitude=longitude
+        )
+
+        if result is None or HITS_KEY not in result:
+            logger.debug(
+                f"No screening location found for coordinates {latitude},{longitude}"
+            )
+            return []
+
+        hits = result[HITS_KEY]
+        logger.debug(
+            f"Found {len(hits)} screening location for coordinates {latitude},{longitude}"
+        )
+        return hits
```

Comment:
You don't seem to make any kind of sorting, but it is mentioned in the issue. Besides, I think we should return a processed output instead of the json result, like a list of instances of a class, with known properties to access. It would be safer than just accessing keys in a dict in the code that will use it.

Reply:
I wasn't sure if we were going to return the result as-is to the frontend and let the frontend render the details as needed. If we are going to use any of those property on the backend, it would indeed be safer to transform the result in some class instead.

@fparga any thoughts?

Reply:
The frontend should receive the content of the carousel from the dialogue, it shouldn't have anything to do with deciding what to do with the results...

Reply:
About the sorting: it is already sorted by the API. Will update the issue.

Reply:
@NetBKarine is right, the frontend won't have anything to do with any of this and will just receive a message in a custom format to display the carousel with the info.

Reply:
Yes sorry - we already discussed that offline. Will fix that, thanks!